### DISCUSSION TITLE
Merge into Electron upstream

### DISCUSSION THREAD

**Discussion by samuelmaddock** [castlabs/electron-releases#24] [Q&A]
Hi, I'm interested in this project and was wondering if there has been any effort to merge these changes into Electron upstream. Have they been receptive to the idea? If not, what kind of blockers are there?

**Comment by khwaaj** [comment]
Hi @samuelmaddock,

Currently there are a few things, unrelated to upstream, preventing us from doing this. Some of these obstacles are going away though, so, at some point, we may reconsider and try to get the updates accepted by upstream (if they'll have them). This needs to be discussed internally though and will likely not happen near-term.

**Reply by samuelmaddock** [reply]
@khwaaj were these obstacles resolved in chromium? I noticed that CEF seemingly added support outside of the required .sig files. 
https://github.com/chromiumembedded/cef/issues/3404

**Comment by samuelmaddock** [comment]
Thanks for the response @khwaaj, please let us know if any developments occur in the future. I know there are a few folks in electron/electron#12427 who would appreciate these changes merged into upstream, myself included.

Out of curiosity, why is the source code for this fork not publicly available?

**Comment by khwaaj** [comment]
Basically it is not open for legacy security reasons, i.e. we currently have some code additions that we can't open-source as part of the solution. In more recent CDMs most of this is no longer necessary, but the code is still in the repo, so...

Anyway, I'll bring this topic up for discussion and I'll try to update this thread if anything new is decided.

**Comment by samuelmaddock** [comment]
@khwaaj Would it be possible for Castlabs to cleanup the Git history or legacy code of the fork to open source it? It'd be really helpful to see the changes required to support DRM-related features. I'm interested in using this fork, but afraid of it falling too far behind Electron or being abandoned one day.

**Comment by khwaaj** [comment]
@samuelmaddock, sorry for not keeping you updated, but this is still an open topic and may happen at some point in the future. There are still several things holding us back though (and it is not just a matter of legacy code as I may have suggested in my comment above), so it is not something I would count on near term.

**Comment by samuelmaddock** [comment]
@khwaaj Thanks for the update. This is no longer needed for my project, but I'm sure other folks would still appreciate it. If at some time in the future Castlabs decides to pursue this, feel free to email me and I can get you in contact with the Electron maintainers.

**Comment by ivanov35** [comment]
Is it intended to maintain this castLab fork in the newest state regarding the newest electron features which can come in the future, e.g. to keep up to date the castLab fork with the latest electron features ?

**Comment by khwaaj** [comment]
Yes, that is absolutely the intention. If you have been monitoring this project for a while you have seen that we have never been very far behind stock Electron on stable and beta releases thus far, and that is how we intend to continue.
